### DISCUSSION TITLE
Plot Nodetree Setup

### DISCUSSION THREAD

**Discussion by Sinasta** [nortikin/sverchok#4587] [Ideas]
This is not a bug report and please feel free to close it instantly if it is not an appropriate channel to ask this question. I would like to plot my nodetree setup showing the structure and connection of the different nodes for documentation reasons. So the way i am doing this right now is to take multiple screenshots of the nodetree, add them together and remove the background. I kinda feel like there should somehow be a way to export the Tree to a plottet svg graph where each box is labeled with the node name and lines showing the node connections. Is there such a possibility?

**Comment by zeffii** [comment]
a super minimalist nodetree->svg is relatively uncomplicated to make, but the individual node uis are a lot more involved. so. super abstract is easy, but users may not be a fan of it.

we did experiments also drawing nodes as svg from their specs https://github.com/nortikin/sverchok/issues/269#issuecomment-52401427 (several posts..) for docs, but it felt incomplete.

but..
https://blender.stackexchange.com/questions/12364/big-screenshot-of-node-tree
suggests this addon:  https://github.com/Yeetus3141/PrintNodes/

**Comment by zeffii** [comment]
here's a `sverchok nodetree to svg` script using `lxml` xml elementtree, it allows for exporting the xml prettyprinted / nicely indented. Other than that the api behaves much the same as the python standard library's `xml.etree.ElementTree`

[`g i s t`](https://gist.github.com/zeffii/b25403ec154366f039e6a1ec87e9d35a)

![image](https://user-images.githubusercontent.com/619340/179921636-b6b0d3f9-6cd2-42fc-bd6a-4e5b68ef613e.png)

**Comment by Sinasta** [comment]
This is just amazing!

**Comment by zeffii** [comment]
I didn't manage to get "Frames" in the right position yet, but that's about the extent of what can be done with relative simplicity. Drawing the ui of the nodes would require a full blown reverse-engineering of Blender's UILayout api - and that's not going to happen.. at least, i won't be doing it.

**Comment by zeffii** [comment]
I think the Blender NodeFrame's  `location` property is broken at the moment (or maybe i'm being stupid), therefore i compute the bounding box and introduce padding.

![image](https://user-images.githubusercontent.com/619340/180566965-cff4fefa-7228-4c56-b0dc-19f7c71f9f5f.png)

**Reply by Durman** [reply]
It's impossible to get frame location via Python API. That was state by some of Blender developers.

**Reply by zeffii** [reply]
good. I thought i was losing my mind

**Comment by zeffii** [comment]
gradient noodles.
![image](https://user-images.githubusercontent.com/619340/180598793-8c77236e-2fb7-4eba-a297-61de1e7ad987.png)

todo set element colors (sockets, text) via css by setting classes/id.

**Comment by zeffii** [comment]
uhoh. some degree of UILayout should be possible to wrap..
![image](https://user-images.githubusercontent.com/619340/180652938-4eedd611-6fc3-414e-a0a5-e28d18ea3cd8.png)
https://docs.blender.org/api/current/info_best_practice.html#user-interface-layout

**Comment by zeffii** [comment]
i'll migrate this to it's own repo..   https://github.com/zeffii/sverchok_to_svg.git
